packages/e2e-test-utils-playwright/tsconfig.json

```diff
@@ -0,0 +1,21 @@
+{
+	"extends": "../../tsconfig.base.json",
+	"compilerOptions": {
+		"incremental": false,
+		"composite": false,
+		"module": "CommonJS",
+		"rootDir": "source",
+		"noEmit": false,
+		"outDir": "build",
+		"sourceMap": true,
+		"declaration": true,
+		"declarationMap": true,
+		"declarationDir": "build-types",
+		"emitDeclarationOnly": false,
+		"allowJs": true,
+		"checkJs": false,
+	},
+	"include": [
+		"source/**/*"
```

Comment:
I renamed `src` to `source`. This is the only way I know to prevent `babel` from building them in `build` and `build-module` folder.

We can't use the default babel config because it will compile playwright's fixture into unsupported syntax:

```js
test.extend({
  page: async ({ page }, use) => {
    await use(page);
  },
});

// Will be compiled down to...

test.extend({
  page: async (_ref1, use) => {
    const { page } = _ref1;
    await use(page);
  },
});
```

They are semantically equal, but it breaks Playwright tooling to lazy-load a fixture.

I couldn't find a way to override custom `browserslist` config per package, so I thought the easiest way might be to use `typescript` instead to compile the source. Since that our tooling will look for every JS/TS file inside any `src` folder, I had to rename it to `source` to bypass that check.

Reply:
I think it's fine to modify the build to ignore some packages. 

I had a look and there doesn't seem to be one way to take into account both the full build and the watch task.

For the watch, the build.js script is executed with a list of files by watch.js:
https://github.com/WordPress/gutenberg/blob/fc595baebb0e61185c3880601009073c41ed7478/bin/packages/build.js#L193-L203

You could either ignore the package completely in `watch.js` or avoid pushing it to the steam in the above code.

For the full build, I think this is the relevant bit, looks pretty easy to add some ignore paths:
https://github.com/WordPress/gutenberg/blob/fc595baebb0e61185c3880601009073c41ed7478/bin/packages/build.js#L203-L243

Reply:
Yeah, that makes sense. I'm just not comfortable with all these hard-coded configurations all over the place I guess 😅 . IIRC, I think @youknowriad has planned to do some refactoring around build steps in the future?

Given all that, I'll convert back to `src` with hard-coded configurations for now.